kafka-streams/src/main/java/Account.java

```diff
@@ -0,0 +1,95 @@
+import com.google.gson.JsonObject;
+
+import java.util.Objects;
+
+public class Account {
+
+    private final String giroNumber;
+    private double balance;
+    private final String currency;
+
+    public Account(String giroNumber, String currency, double balance) {
+        this.giroNumber = giroNumber;
+        this.currency = currency;
+        this.balance = balance;
+    }
+
+    public void refresh(Account newAccount) {
+        if (giroNumber.equals(newAccount.giroNumber)) {
```

Comment:
Why are you checking here giroNumber here?

Reply:
Safety measure, to prevent someone from refreshing an account's balance with an inapropriate one. refresh() can be called from the main method as well, and although this condition is checked there as well, I thought this way is safer.

Reply:
Okey, got it